integration_test/ops_agent_test.go

```diff
@@ -465,48 +464,59 @@ func TestHTTPRequestLog(t *testing.T) {
-		// Write a log with an http request in the message to cloud logging and immediately query it.
-		writeAndQueryLog := func(httpRequestKey string, logId string) *cloudlogging.Entry {
-			logBody := map[string]interface{}{
-				"logId":        logId,
-				httpRequestKey: httpRequestBody,
-			}
-			logBytes, err := json.Marshal(logBody)
-			if err != nil {
-				t.Fatalf("could not marshal test log: %v", err)
-			}
-			err = gce.UploadContent(
-				ctx,
-				logger,
-				vm,
-				strings.NewReader(string(logBytes)+"\n"),
-				logPath)
-			if err != nil {
-				t.Fatalf("error writing log line: %v", err)
-			}
+		// Log with HTTP request data nested under "logging.googleapis.com/httpRequest".
+		const newHTTPRequestKey = confgenerator.HttpRequestKey
+		const newHTTPRequestLogId = "new_request_log"
+		newLogBody := map[string]interface{}{
+			"logId":           newHTTPRequestLogId,
+			newHTTPRequestKey: httpRequestBody,
+		}
+		newLogBytes, err := json.Marshal(newLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal new test log: %v", err)
+		}
+
+		// Log with HTTP request data nested under "logging.googleapis.com/http_request".
+		const oldHTTPRequestKey = "logging.googleapis.com/http_request"
+		const oldHTTPRequestLogId = "old_request_log"
+		oldLogBody := map[string]interface{}{
+			"logId":           oldHTTPRequestLogId,
+			oldHTTPRequestKey: httpRequestBody,
+		}
+		oldLogBytes, err := json.Marshal(oldLogBody)
+		if err != nil {
+			t.Fatalf("could not marshal old test log: %v", err)
+		}
+
+		// Write both logs to log source file at the same time.
+		err = gce.UploadContent(
+			ctx,
+			logger,
+			vm,
+			strings.NewReader(fmt.Sprintf("%s\n%s\n", string(newLogBytes), string(oldLogBytes))),
+			logPath)
+		if err != nil {
+			t.Fatalf("error writing log line: %v", err)
+		}
 
+		// Test that the new documented field, "logging.googleapis.com/httpRequest", will be
+		// parsed as expected by Fluent Bit.
+		t.Run("parse new HTTPRequest key", func(t *testing.T) {
+			t.Parallel()
 			entry, err := gce.QueryLog(
 				ctx,
 				logger.ToMainLog(),
 				vm,
 				"mylog_source",
 				time.Hour,
-				fmt.Sprintf("jsonPayload.logId=%q", logId),
+				fmt.Sprintf("jsonPayload.logId=%q", newHTTPRequestLogId),
 				gce.QueryMaxAttempts)
 			if err != nil {
-				t.Fatalf("could not find written log with id %s: %v", logId, err)
+				t.Fatalf("could not find written log with id %s: %v", newHTTPRequestLogId, err)
 			}
-			return entry
-		}
-
-		// Test that the new documented field, "logging.googleapis.com/httpRequest", will be
-		// parsed as expected by Fluent Bit.
-		t.Run("parse new HTTPRequest key", func(t *testing.T) {
-			httpRequestKey := confgenerator.HttpRequestKey
-			entry := writeAndQueryLog(httpRequestKey, "341231")
 			payload := entry.Payload.(*structpb.Struct)
 			for key := range payload.GetFields() {
-				if key == httpRequestKey {
+				if key == newHTTPRequestKey {
 					t.Fatal("expected request key to be stripped out of message")
```

Comment:
Minor nit: this is a payload, not a message. Also, we should report which key was expected to be stripped out.
```suggestion
					t.Fatalf("expected %s key to be stripped out of the payload", newHTTPRequestKey)
```

Reply:
Changed in [dd1c501](https://github.com/GoogleCloudPlatform/ops-agent/pull/655/commits/dd1c501d2202c9a0aff9c66528753ebae6447ce2)